internal/provider/provider.go

```diff
@@ -150,6 +151,28 @@ func New() *schema.Provider {
 					},
 				},
 			},
+			"coder_host": {
```

Comment:
nit: I'm wondering now if `coder_host` is the best term for this because this technically checks the provisioner's environment, right? 

By default, the Coder host, provisioner, (and docker socket) all run in the same environment, so the outcome is still positive for Docker template users.

Deployments that eventually add eventual provisioners (and/or docker hosts) will understand the separation of concerns so they won't really need to dynamically fetch the host's environment, but there _may_ be some merit in understanding which architecture the provisioner is on? (which this does, but it calls it the host). Still an obscure use case and it's really meant to help with single-machine deployments.

I don't really have a problem with it being called coder_host, just thinking out loud and maybe we can link to this comment in the future if it comes up again.

Reply:
I think it is the host of the provisioner though, or at least that's what I'm intending for it to portray. 

We could alternatively do `coder_provisioner_host` or `coder_provisioner`. Maybe `coder_provisioner` would be better, because it seems likely we'll expose other options from that eventually :thinking:. Thoughts?

Reply:
I changed it to `coder_provisioner`. It makes more sense IMO.